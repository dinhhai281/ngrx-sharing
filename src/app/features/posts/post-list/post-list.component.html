<ng-container *ngIf="vm$ | async as vm">
  <div class="grid grid-cols-4 gap-4 pt-2">
    <mat-card *ngFor="let post of vm.posts">
      <mat-card-title class="truncate text-pink-700">
        {{ post.title }}
      </mat-card-title>

      <mat-divider></mat-divider>

      <mat-card-content class="h-48 py-2">
        {{ post.body }}
      </mat-card-content>

      <mat-divider></mat-divider>

      <mat-card-actions align="end">
        <button
          mat-button
          class="focus:outline-none"
          color="primary"
          [disabled]="vm.disableAdd"
          [matMenuTriggerFor]="menu"
          [matMenuTriggerData]="{ postId: post.id }"
        >
          Add to collections
        </button>
      </mat-card-actions>
    </mat-card>
  </div>

  <mat-menu #menu="matMenu" xPosition="before">
    <ng-template matMenuContent let-postId="postId">
      <button
        mat-menu-item
        *ngFor="let collection of vm.collections"
        class="focus:outline-none"
        (click)="onAddPostToCollection(postId, collection.id)"
      >
        {{ collection.name }}
      </button>
    </ng-template>
  </mat-menu>
</ng-container>
