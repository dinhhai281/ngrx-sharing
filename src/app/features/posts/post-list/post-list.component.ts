import {
  Component,
  OnInit,
  ViewEncapsulation,
  ChangeDetectionStrategy,
  Input,
  Output,
  EventEmitter,
} from '@angular/core';
import { Collection, Post } from '@app/models';

@Component({
  selector: 'app-post-list',
  templateUrl: './post-list.component.html',
  styleUrls: ['./post-list.component.scss'],
  encapsulation: ViewEncapsulation.None,
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class PostListComponent implements OnInit {
  @Input() posts: Post[];
  @Input() collections: Collection[];
  @Output('selectPost') select$ = new EventEmitter<number>();
  @Output('addToCollection') collectionClick$ = new EventEmitter<{
    postId: number;
    collectionId: string;
  }>();

  constructor() {}

  ngOnInit() {}
}
