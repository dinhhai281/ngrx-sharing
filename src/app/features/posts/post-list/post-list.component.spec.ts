import { MatCardModule } from '@angular/material/card';
import { MatDividerModule } from '@angular/material/divider';
import { render, RenderResult } from '@testing-library/angular';

import { PostListComponent } from './post-list.component';

describe('PostListComponent', () => {
  let component: RenderResult<PostListComponent>;

  beforeEach(async () => {
    component = await render(PostListComponent, {
      imports: [MatCardModule, MatDividerModule],
      componentProperties: {
        posts: [{ title: 'title', id: 1, userId: 1, body: 'body' }],
        collections: [{ name: 'collection', posts: [], userId: 'userid', id: 'test-collection' }],
      },
    });
  });

  it('should match snapshot', () => {
    expect(component.container).toMatchSnapshot();
  });
});
