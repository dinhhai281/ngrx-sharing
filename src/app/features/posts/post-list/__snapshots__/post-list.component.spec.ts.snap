// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostListComponent should match snapshot 1`] = `
<div
  id="root0"
>
  <div
    class="grid grid-cols-4 gap-4 pt-2"
  >
    
    <mat-card
      class="mat-card mat-focus-indicator _mat-animation-noopable"
    >
      <mat-card-title
        class="truncate text-pink-700 mat-card-title"
      >
         title 
      </mat-card-title>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
      <mat-card-content
        class="h-48 py-2 mat-card-content"
      >
         body 
      </mat-card-content>
      <mat-divider
        aria-orientation="horizontal"
        class="mat-divider mat-divider-horizontal"
        role="separator"
      />
      <mat-card-actions
        align="end"
        class="mat-card-actions mat-card-actions-align-end"
      >
        <button
          class="focus:outline-none"
          color="primary"
          mat-button=""
        >
           Add to collections 
        </button>
      </mat-card-actions>
    </mat-card>
  </div>
</div>
`;
